backend/src/test/java/com/barmjz/productivityapp/registration/RegistrationServiceTest.java

```diff
@@ -0,0 +1,194 @@
+        registrationService = new RegistrationService(userService, emailValidator, passwordValidator, passwordEncoder);
+    }
+
+    @Test
+    void ThrowsWhenEmailNotValid(){
+        // Given
+        String email = "dummy";
+        String password = "password";
+        String firstName = "Jack";
+        String lastName = "Grealish";
+        RegistrationRequest registrationRequest = new RegistrationRequest(email,
+                password,
+                firstName,
+                lastName);
+        given(emailValidator.test(email)).willReturn(false);
+
+        // Then
+        assertThatThrownBy( () ->  registrationService.register(registrationRequest))
+                .isInstanceOf(IllegalStateException.class)
+                .hasMessageContaining("invalid email");
+        verify(userService, never()).saveUser(any(), any());
+
+    }
+
+    @Test
+    void ThrowsWhenPasswordNotValid(){
+        // Given
+        String email = "dummy@gmail.com";
+        String password = "123";
+        String firstName = "Jack";
+        String lastName = "Grealish";
+        RegistrationRequest registrationRequest = new RegistrationRequest(email,
+                password,
+                firstName,
+                lastName);
+        given(emailValidator.test(email)).willReturn(true);
+        given(passwordValidator.test(password)).willReturn(false);
+
+        // Then
+        assertThatThrownBy( () ->  registrationService.register(registrationRequest))
+                .isInstanceOf(IllegalStateException.class)
+                .hasMessageContaining("weak password");
+        verify(userService, never()).saveUser(any(), any());
+
+    }
+    @Test
+    void ThrowsWhenFirstNameIsNull(){
+        // Given
+        String email = "dummy@gmail.com";
+        String password = "123";
+        String firstName = null;
+        String lastName = "Grealish";
+        RegistrationRequest registrationRequest = new RegistrationRequest(email,
+                password,
+                firstName,
+                lastName);
+
+        // Then
+        assertThatThrownBy( () ->  registrationService.register(registrationRequest))
+                .isInstanceOf(NullPointerException.class)
+                .hasMessageContaining("first and last name can't be null");
+        verify(userService, never()).saveUser(any(), any());
+
+    }
+
+    @Test
+    void ThrowsWhenLastNameIsNull(){
+        // Given
+        String email = "dummy@gmail.com";
+        String password = "123";
+        String firstName = "Jack";
+        String lastName = null;
+        RegistrationRequest registrationRequest = new RegistrationRequest(email,
+                password,
+                firstName,
+                lastName);
+
+        // Then
+        assertThatThrownBy( () ->  registrationService.register(registrationRequest))
+                .isInstanceOf(NullPointerException.class)
+                .hasMessageContaining("first and last name can't be null");
```

Comment:
I can't find an error message containing "first and last name can't be null" int the source code

Reply:
I added it in the RegistrationService Class in the register method.

Reply:
![image](https://user-images.githubusercontent.com/73697437/205453779-6aa29f1a-5acc-47c6-bbc0-80c9b403e3e8.png)
Here

Reply:
I have found it.